maven-core/src/main/java/org/apache/maven/internal/transformation/ConsumerPomArtifactTransformer.java

```diff
@@ -77,13 +80,34 @@ public void injectTransformedArtifacts(MavenProject project, RepositorySystemSes
                 Files.createDirectories(buildDir);
                 generatedFile = Files.createTempFile(buildDir, CONSUMER_POM_CLASSIFIER, "pom");
             }
+
+            removeOldConsumerPomFiles(generatedFile);
+
             project.addAttachedArtifact(new ConsumerPomArtifact(project, generatedFile, session));
         } else if (project.getModel().isRoot()) {
             throw new IllegalStateException(
                     "The use of the root attribute on the model requires the buildconsumer feature to be active");
         }
     }
 
+    private void removeOldConsumerPomFiles(Path generatedFile) throws IOException {
+        List<Path> oldConsumerPomFiles;
```

Comment:
Why does the code collect all the files in a list if it could as well remove them in the terminal operation of the `j.u.s.Stream`?

Reply:
I had a couple of reasons for it. The first reason for me is readability (as I'll show down below). 

The second is that I don't think its a good practice to wrap the checked exception into a runtime exception (also introducing more code or a wrapper 'delete' method). Especially since `injectTransformedArtifacts()` already throws a checked exception which is probably accounted for somewhere at some point.

As far as I know when you rewrite the stream to something like `stream.filter(path -> /*filter*/ }).forEach(Files::delete);` you either have to wrap the checked exception thrown by `Files.delete(path)` into a runtime exception:
```java
    private void delete( Path path ) {
        try { Files.delete( path ); }
        catch ( IOException e ) { throw new RuntimeException("", e); }
    }
    
    /* obfuscated*/
    
    stream.filter(path -> /*filter*/ })
           .forEach(this::delete);
```
Or handle it:
```java
stream.filter(path -> /*filter*/ })
           .forEach(path -> {
                try { Files.delete(path); }
                catch (IOException ex) { /*handle the exception*/ }
           });
```

Reply:
That makes perfect sense to me. Thanks for commenting on this!